pkg/bff/organization.go

```diff
@@ -13,6 +15,145 @@ import (
 	storeerrors "github.com/trisacrypto/directory/pkg/store/errors"
 )
 
+// CreateOrganization creates a new organization in the database. This endpoint returns
+// an error if the organization already exists and the user is assigned to it. The user
+// must have the create:organizations permission to perform this action.
+func (s *Server) CreateOrganization(c *gin.Context) {
+	var (
+		err  error
+		user *management.User
+	)
+
+	// Fetch the user from the context
+	if user, err = auth.GetUserInfo(c); err != nil {
+		log.Error().Err(err).Msg("create organization handler requires user info; expected middleware to return 401")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not identify user to create organization"))
+		return
+	}
+
+	// Load the user app metadata to check their organization assignments
+	appdata := &auth.AppMetadata{}
+	if err = appdata.Load(user.AppMetadata); err != nil {
+		log.Error().Err(err).Msg("could not parse user app metadata")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not parse user app metadata"))
+		return
+	}
+
+	// Unmarshal the params from the POST request
+	params := &api.OrganizationParams{}
+	if err := c.ShouldBind(params); err != nil {
+		log.Warn().Err(err).Msg("could not bind request")
+		c.JSON(http.StatusBadRequest, &api.Reply{Error: err.Error()})
```

Comment:
```suggestion
		c.JSON(http.StatusBadRequest, api.ErrorResponse(err))
```

You can use `api.ErrorResponse` with `errors` as well if you'd like - it's a bit simpler.

Reply:
Thanks, I forgot about that one.